src/backend/distributed/transaction/worker_transaction.c

```diff
@@ -511,6 +541,48 @@ SendOptionalCommandListToWorkerInTransaction(const char *nodeName, int32 nodePor
 }
 
 
+/*
+ * SendOptionalCommandListToWorkerInCoordinatedTransaction sends the given
+ * command list to the given worker as part of the coordinated transaction.
+ * If any of the commands fail, the function returns false.
+ */
+bool
+SendOptionalCommandListToWorkerInCoordinatedTransaction(const char *nodeName, int32
+														nodePort,
+														const char *nodeUser,
+														List *commandList)
+{
+	int connectionFlags = 0;
+	bool failed = false;
+
+	UseCoordinatedTransaction();
+
+	MultiConnection *workerConnection =
+		GetNodeUserDatabaseConnection(connectionFlags, nodeName, nodePort,
+									  nodeUser, NULL);
+	if (PQstatus(workerConnection->pgConn) != CONNECTION_OK)
+	{
+		return false;
+	}
+
+	RemoteTransactionsBeginIfNecessary(list_make1(workerConnection));
+
+	/* iterate over the commands and execute them in the same connection */
+	const char *commandString = NULL;
+	foreach_ptr(commandString, commandList)
+	{
+		if (ExecuteOptionalRemoteCommand(workerConnection, commandString, NULL) !=
+			RESPONSE_OKAY)
+		{
+			failed = true;
```

Comment:
What happens after this? 

Do we need a MarkRemoteTransactionFailed(connection, false); at least?

Reply:
I think that makes sense. I think such a change, (a) we `ROLLBACK` any changes at COMMIT time (b) Keep the connection open so that if another command is executed, it'd fail as long as  is not optional